Sources/ArgumentParser/Parsable Properties/ArgumentHelp.swift

```diff
@@ -24,9 +35,21 @@ public struct ArgumentHelp {
   ///   flags don't include a value.
   public var valueName: String?
   
+  /// A visibility level indicating whether this argument should be shown in
+  /// the extended help display.
+  public var visibility: Visibility = .default
+
   /// A Boolean value indicating whether this argument should be shown in
   /// the extended help display.
-  public var shouldDisplay: Bool = true
+  @available(*, deprecated, message: "Use visibility level instead.")
+  public var shouldDisplay: Bool {
+    get {
+      return visibility == .default
+    }
+    set {
+      visibility = newValue ? .default : .hidden
+    }
+  }
   
   /// Creates a new help instance.
   public init(
```

Comment:
Could we add a depreciation to this initializer start moving away from `shouldDisplay`

Reply:
Sure!

Reply:
I marked this initializer as deprecated but didn't provide a deprecation message since I don't any a public alternative for now. However, I just learned (from you :)) about `@_disfavoredOverload`, do you think it could be useful here?

Reply:
I think the message should guide users to `ArgumentHelp.init(_:discussion:valueName: visibility:)`. `@_disfavoredOverload` probably is needed as well since both inits can be called without providing any arguments.

Reply:
That's a clever solution!